app/static/js/index.js

```diff
@@ -356,6 +377,20 @@ function loadNextPage() {
     }
 }
 
+function updateCriteria(criteria) {
+    let urlSearchParams = new URLSearchParams(window.location.search);
+    let params = Object.fromEntries(urlSearchParams.entries());
+    if (params['query'] != 'undefined') {
+        criteria['query'] = params['query'];
+    }
+    for (const [param, paramValue] of Object.entries(params)) {
+        if (param != 'query') {
```

Comment:
Perhaps see if we can pop the query from the `params` obj so that you don't have to explicitly exclude it here